test/unit/deferred.js

```diff
@@ -146,30 +150,70 @@ test( "jQuery.Deferred.then - filtering (fail)", function() {
 		value2 = b;
 	});
 
-	defer.reject( 2, 3 );
-
-	strictEqual( value1, 2, "first reject value ok" );
-	strictEqual( value2, 3, "second reject value ok" );
-	strictEqual( value3, 6, "result of filter ok" );
+	defer.reject( 2, 3 ).then( null, function() {
+		assert.strictEqual( value1, 2, "first reject value ok" );
+		assert.strictEqual( value2, 3, "second reject value ok" );
+		assert.strictEqual( value3, 6, "result of filter ok" );
+		done.pop().call();
+	});
 
 	jQuery.Deferred().resolve().then( null, function() {
-		ok( false, "then should not be called on resolve" );
+		assert.ok( false, "then should not be called on resolve" );
+	}).then( done.pop() );
+
+	jQuery.Deferred().reject().then( null, jQuery.noop ).done(function( value ) {
+		assert.strictEqual( value, undefined, "then fail callback can return undefined/null" );
+		done.pop().call();
+	});
+});
+
+test( "[DIVERGENT] jQuery.Deferred.pipe - filtering (fail)", function( assert ) {
+
+	assert.expect( 4 );
+
+	var value1, value2, value3,
+		defer = jQuery.Deferred(),
+		piped = defer.pipe( null, function( a, b ) {
+			return a * b;
+		}),
+		done = jQuery.map( new Array( 3 ), function() { return assert.async(); } );
+
+	piped.fail(function( result ) {
```

Comment:
That piped deferreds _inherit_ rejection is the reason why I couldn't justify dropping `.pipe`... the standard and our current behavior are just too divergent here.